be-bop-wizard/be-bop-wizard.sh

```diff
@@ -874,6 +894,28 @@ validate_bebop_config_domain() {
     return 0
 }
 
+# Build list of domains for SSL certificates and nginx configuration
+# Rules:
+#   www.domain.com → domain.com, www.domain.com, s3.domain.com
+#   domain.com → domain.com, s3.domain.com (no www)
+#   app.domain.com → app.domain.com, s3.app.domain.com (s3 prefix on subdomain)
+build_domain_list() {
+    local input_domain="$1"
+    local parts=$(echo "$input_domain" | tr '.' '\n' | wc -l)
+
+    if [[ "$input_domain" =~ ^www\. ]]; then
+        # User specified www.domain.com - include root, www, and s3
+        local root_domain="${input_domain#www.}"
+        echo "$root_domain $input_domain s3.$root_domain"
+    elif [[ $parts -eq 2 ]]; then
+        # User specified root domain - include only root and s3 (no www)
+        echo "$input_domain s3.$input_domain"
+    else
+        # User specified other subdomain - s3 should be prefixed to subdomain
+        echo "$input_domain s3.$input_domain"
+    fi
```

Comment:
This snippet evaluates the specified domain and **decides** what domain to use for s3, what domain to use for be-BOP and what extra domains to configure nginx for.

Because this is a decision, it needs to happen at a different place (perhaps in `inspect_system_state`), record the decisions as facts (e.g. `SYSTEM_STATE+=("bebop_domain=...")`) and be reported to the user (perhaps in `summarize_state_and_plan`).